docs/spl-docs/time-modifiers.md

```diff
@@ -0,0 +1,177 @@
+| quarters   | `q`, `qtr`, `qtrs`, `quarter`, `quarters`     |
+| years      | `y`, `yr`, `yrs`, `year`, `years`             |
+
+When specifying relative time, use `now` to refer to the current time.
+
+### Relative time modifiers that snap to a time
+
+In relative time settings, you can specify a snap to time, which acts as an offset from the relative time. This snap to time unit will round down to the nearest or latest time for the time amount that you specify. To apply this, simply separate the time amount from the snap to time unit using an `@` symbol.
+
+The syntax for the snap to time unit is `[+|-]<time_integer><time_unit>@<time_unit>`.
+
+When snapping to the nearest or latest time, the system always snaps backwards or rounds down to the latest time that is not after the specified time. For example, the current time is 15:45:00 and the snap to time is earliest=-h@h. The time modifier snaps to 14:00.
+
+You can also define the relative time modifier using only the snap to time unit. For example, to snap to a specific day of the week, use @w0 for Sunday, @w1 for Monday, and so on. For Sunday, you can specify either w0 or w7.
+
+When a snap to time unit isn't defined, the system defaults to using seconds. If an invalid snap to time unit is provided, it may result in a syntax error or unexpected search results.
+
+The snap to option becomes very useful in a range of situations. For example, if you want to search for events in the previous month, specify `earliest=-mon@mon latest=@mon`. This sets the search to start at the beginning of the last month and end at the beginning of this month.
+
+### Difference between relative time and relative snap to time
+
+Imagine it's May 15th, and you initiate a search at 10:30 AM.
+
+- If you use earliest=-3h, the search covers the past three hours, beginning at 7:30 AM on May 15th.
+- If you use earliest=-3h@h, the search adjusts to three hours ago but snaps to the start of the hour. This means the search includes events from 7:00 AM on May 15th.
+
+### Special time units
+
+The following abbreviations are reserved for special cases of time units and snap time offsets.
+
+
+
+| Time Unit | Description |
+|-----------|-------------|
+| `earliest=1` | To search events from the start of UNIX epoch time, use `earliest=1`. UNIX epoch time 1 is UTC January 1, 1970, at 12:00:01 AM. |
+| `earliest=0` | Including `earliest=0` in the search string indicates that time is not considered in the search. |
+| `earliest=1` and `latest=now` or `latest=<a_large_number>` | The search spans all time. However, specifying `latest=now` (the default) excludes future events, while `latest=<a_large_number>` includes future events, which are events with timestamps beyond the current time, `now()`. |
+| `earliest=now` or `latest=now` | Specifies that the search starts or ends at the current time. Any search including `earliest=<relative time offset>` should also include `latest=now`. For example, `earliest=-30s latest=now`. |
```

Comment:
> Any search including `earliest=<relative time offset>` should also include `latest=now`.

But on line 37 you say:
> If you specify only the earliest time modifier, latest is set to the current time `now` by default.

So if I include `earliest`, should I set `latest=now`, or does it not matter? Or does it only matter if I set `earliest` using a relative time modifier as opposed to absolute?

Reply:
I just verified this again on siglens, we are automatically setting latest to `now` irrespective of whether its a relative time modifier or absolute. I updated the description in the latest commit.